File: frontend/src/constants/notification.ts
// 通知类型枚举（与后端 backend/internal/constants/notification.go 保持一致）
export const NotificationType = {
  SIGNUP_SUCCESS: 'signup_success',
  REMINDER: 'reminder',
  REVIEW_RESULT: 'review_result',
  CHECKIN_SUCCESS: 'checkin_success',
} as const

export const NotificationTypeText: Record<string, string> = {
  [NotificationType.SIGNUP_SUCCESS]: '报名成功',
  [NotificationType.REMINDER]: '活动提醒',
  [NotificationType.REVIEW_RESULT]: '审核结果',
  [NotificationType.CHECKIN_SUCCESS]: '签到成功',
}

export const UserRole = {
  USER: 'user',
  ORGANIZER: 'organizer',
  ADMIN: 'admin',
} as const

export const UserRoleText: Record<string, string> = {
  [UserRole.USER]: '普通用户',
  [UserRole.ORGANIZER]: '组织者',
  [UserRole.ADMIN]: '管理员',
}

File: backend/internal/constants/notification.go
package constants

// NotificationType 消息通知类型枚举。
const (
	NotificationSignupSuccess  = "signup_success"
	NotificationReminder       = "reminder"
	NotificationReviewResult   = "review_result"
	NotificationCheckinSuccess = "checkin_success"
)

// NotificationTypeValues 全部通知类型值。
var NotificationTypeValues = []string{NotificationSignupSuccess, NotificationReminder, NotificationReviewResult, NotificationCheckinSuccess}

// UserRole 用户角色枚举。
const (
	RoleUser      = "user"
	RoleOrganizer = "organizer"
	RoleAdmin     = "admin"
)

// UserRoleValues 全部角色值。
var UserRoleValues = []string{RoleUser, RoleOrganizer, RoleAdmin}
